docs/_pages/grammar.md

```diff
@@ -63,17 +63,21 @@ SimpleType =
     NAME ['.' NAME] [ '<' [TypeParams] '>' ] |
     'typeof' '(' exp ')' |
     TableType |
-    FunctionType
+    FunctionType |
+    '(' SimpleType ')'
 
 SingletonType = STRING | 'true' | 'false'
 
-Type =
-    SimpleType ['?'] |
-    Type ['|' Type] |
-    Type ['&' Type]
+UnionSuffix = {'?'} ['|' SimpleType]
+IntersectionSuffix = ['&' SimpleType]
+Type = SimpleType {UnionSuffix | IntersectionSuffix}
+
+GenericTypePackParameter = NAME '...'
+GenericTypeList = NAME [',' GenericTypeList] | GenericTypePackParameter {',' GenericTypePackParameter}
+
+GenericTypePackParameterWithDefault = NAME '...' ['=' (TypePack | VariadicTypePack | GenericTypePack)]
+GenericTypeListWithDefaults = NAME ['=' Type] [',' GenericTypeListWithDefaults] | GenericTypePackParameterWithDefault {',' GenericTypePackParameterWithDefault}
```

Comment:
I think this is still actually slightly wrong. It does not encode that after one generic is specified with a default, then all other generics must also have a default. It currently allows something like `<X = string, Y, Z... = ...string, P...>`.

Something like the following may be better:
```suggestion
GenericTypePackParameterWithDefault = NAME '...' '=' (TypePack | VariadicTypePack | GenericTypePack)
GenericTypeListWithDefaults = 
    GenericTypeList |
    NAME {',' NAME} {',' NAME '=' Type} {',' GenericTypePackParameterWithDefault} |
    NAME '=' Type {',' GenericTypePackParameterWithDefault} |
    GenericTypePackParameterWithDefault {',' GenericTypePackParameterWithDefault}
```
But is it the simplest it can be? I think it covers everything